src/sync/sync_manager.hpp

```diff
@@ -155,8 +156,30 @@ friend class SyncSession;
     // Protects m_users
     mutable std::mutex m_user_mutex;
 
-    // A map of user identities to (shared pointers to) SyncUser objects.
-    std::unordered_map<std::string, std::shared_ptr<SyncUser>> m_users;
+    // A custom key for the user map system. It's basically a hashable tuple.
+    struct UserMapKey {
```

Comment:
Does this represent the same information as `StringPair` above? If so, they should probably just be the same type.